Scripts/ICC: Minor Fixes for Lich King

Demonid opened the pull request:
**Changes proposed:**

- Fixed Shadow Trap and Defile Targets
- Fixed Shadow Trap Summon Delay and Duration
- Fixed Berserk Timer (15 Minutes)
- Fixed Shadow Trap Despawn when Snowstorm starts

I don't have Sniffs but everything on this PR can be check on this video:

https://www.youtube.com/watch?v=V0NNgcC--RY

**Target branch(es):**

3.3.5
master

Grimton1337 commented:
Necrotic Plague should NOT target tanks. Though, it can jump unto the tank, but in order to cast Necrotic Plague, there must be a secondary target (I think a pet can be targeted, but not sure).

Keader commented:
He dont changed necrotic plague...

Demonid, changing place of Shadow Trap despawn, realy change something?

Demonid commented:
@Keader The Necrotic Plague stuff was on Todo List, but was removed after checking some videos + his comment.

The Shadow Trap Despawn change makes the despawn at the exact moment it does on retail, as soon as the snowstorm starts, right now is way before or way after i don't actually remember but you can test it out if you want.

Demonid commented:
Ok, this is all im gonna add to this PR.

@sirikfoll I want to talk to you about a second PR to make the encounter even more Blizzlike.

Keader commented:
This delay stuff can be checked in retail,  i will try sniff it before travel today.
Sirikfoll confirmed some modifications of this PR with sniffs, but he cant confirm if 73540 has 60 sec of duration like you changed (maybe i can get some information when sniff it)
And you checked what happens with valkyr (on tc) when she cant grap the tanker? if she start attack him or stay "passive" like on video?

Demonid commented:
@Keader 73540 case is easy to verify.

On minute 4:10, Shadow Trap the first Shadow Trap is Fully Activated: https://youtu.be/V0NNgcC--RY?t=4m10s

On minute 5:10 you can see the first Shadow Trap despawn: https://youtu.be/V0NNgcC--RY?t=5m8s

So i don't think anymore is needed for this to be verified, also, on retail you never see more than 4 Shadow Traps at the same time, right now on TC you can have up to 5 activated.

Last time i checked, Valkyr stays on standby until a proper target can be found. (Second Target).

Keader commented:
Hmm, looks correct to me :)

Keader commented on src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp:
```diff
@@ -815,6 +815,7 @@ class boss_the_lich_king : public CreatureScript
                 {
                     me->GetMap()->SetZoneOverrideLight(AREA_ICECROWN_CITADEL, LIGHT_SNOWSTORM, 5000);
                     me->GetMap()->SetZoneWeather(AREA_ICECROWN_CITADEL, WEATHER_STATE_LIGHT_SNOW, 0.5f);
+                    summons.DespawnEntry(NPC_SHADOW_TRAP);
```
@Demonid SpellHitTarget is called when spell hit a target right?
So it will call Despawn Shadow Trap alot of times no? (same to maps modifications)

Demonid replied:
I believe it will only be called 2 times, one time for each Remorless Winter since the despawn is under a if: 

if (spell->Id == REMORSELESS_WINTER_1 || spell->Id == REMORSELESS_WINTER_2)

Keader replied:
Naah, you are correct.
Both spells is Self cast with SPELL_AURA_PERIODIC_TRIGGER_SPELL  effect
It will hit only Lich King, so will be called 1 time per cast

Shauren replied:
pls, you think that such a major failure would be present in the original script?

Keader commented:
@Demonid PR is ready? can be merged?

Demonid committed:
Scripts/ICC: Minor Fixes for Lich King

- Fixed Shadow Trap and Defile Targets
- Fixed Shadow Trap Summon Delay and Duration
- Fixed Berserk Timer (15 Minutes)
- Fixed Shadow Trap Despawn when Snowstorm starts

Demonid commented:
@Keader Its ready if no issues are found.

Keader commented:
You dont need squash commits anymore, we can do it while merge :)
